### PR TITLE
Processing chain editor

### PR DESCRIPTION
**Pull request by SteRiccio**

### CODE REVIEW

File: common/analysis/processingStepCalculation.js [hunk 1]
```diff
@@ -1,34 +1,72 @@
...
+  cnt: 'cnt',
   min: 'min',
-  sum: 'sum',
+  max: 'max',
+  med: 'med',
 }
+
+// ====== CREATE
+
+export const newProcessingStepCalculation = (processingStepUuid, index) => ({
```
Comment (minotogna) [inline common/analysis/processingStepCalculation.js]: this method is already available in ProcessingChain

File: core/i18n/resources/en.js [hunk 2]
```diff
@@ -340,6 +341,26 @@ export const enTranslation = {
...
+    aggregateFunction: 'Aggregate function',
+    aggregateFunctions: {
+      avg: 'Average',
+      cnt: 'Count',
+      max: 'Maximum',
+      med: 'Median',
+      min: 'Minimum',
+      sum: 'Sum',
+    },
+    confirmCancel: `**There are unsaved changes**.
```
Comment (minotogna) [inline core/i18n/resources/en.js]: should we add this to common? (to be used by nodeDefView as well?)

File: server/modules/analysis/repository/processingStepCalculationRepository.js [hunk 3]
```diff
@@ -58,4 +66,20 @@ export const updateCalculationIndex = async (surveyId, processingStepUuid, index
   return queriesRes[queriesRes.length - 1]
 }
 
+export const updateCalculation = async (surveyId, calculation, client = db) =>
```
Comment (minotogna) [inline server/modules/analysis/repository/processingStepCalculationRepository.js]: i would use the same convention than insert. ```updateCalculationStep```

File: common/activityLog/activityLog.js [hunk 4]
```diff
@@ -60,6 +60,7 @@ export const type = {
   processingStepDelete: 'processingStepDelete',
   processingStepCalculationCreate: 'processingStepCalculationCreate',
   processingStepCalculationIndexUpdate: 'processingStepCalculationIndexUpdate',
+  processingStepCalculationUpdate: 'processingStepCalculationUpdate',
```
Comment (minotogna) [inline common/activityLog/activityLog.js]: is the activity log in dashboard home show the right content ?

File: core/i18n/resources/en.js (hunk 2 above)
Comment (SteRiccio) [reply core/i18n/resources/en.js]: right

File: server/modules/analysis/repository/processingStepCalculationRepository.js (hunk 3 above)
Comment (SteRiccio) [reply server/modules/analysis/repository/processingStepCalculationRepository.js]: or updateStepCalculation?

File: common/activityLog/activityLog.js (hunk 4 above)
Comment (SteRiccio) [reply common/activityLog/activityLog.js]: no, it doesn't, I'll fix it

File: server/modules/analysis/repository/processingStepCalculationRepository.js (hunk 3 above)
Comment (SteRiccio) [reply server/modules/analysis/repository/processingStepCalculationRepository.js]: ok

File: common/analysis/processingStepCalculation.js (hunk 1 above)
Comment (SteRiccio) [reply common/analysis/processingStepCalculation.js]: right, I'll remove it

**Pull request merged**